aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/strategy/sampling/pollers/TargetPoller.java

```diff
@@ -24,21 +25,30 @@
     private AWSXRay client;
     private Clock clock;
     private CentralizedManifest manifest;
+    private ScheduledExecutorService executor;
 
     public TargetPoller(CentralizedManifest m, AWSXRay client, Clock clock) {
         this.manifest = m;
         this.client = client;
         this.clock = clock;
+        executor = Executors.newSingleThreadScheduledExecutor();
     }
 
     public void start() {
-    Executors.newSingleThreadScheduledExecutor().scheduleAtFixedRate(() -> {
+        executor.scheduleAtFixedRate(() -> {
             try {
                 pollManifest();
-            } catch (Exception ex) {
-                logger.error("Encountered error polling GetSamplingTargets: ", ex);
+            } catch (Throwable t) {
+                logger.error("Encountered error polling GetSamplingTargets: ", t);
+                // An Error should not be handled by the application.
+                // The executor will die and not abrupt main thread.
+                if(t instanceof Error) { throw t; }
             }
-        }, 0, getJitterInterval(), TimeUnit.MILLISECONDS);
+        }, PERIOD * 1000, getJitterInterval(), TimeUnit.MILLISECONDS);
+    }
+
+    public void shutdown() {
+        executor.shutdownNow();
     }
 
     private void pollManifest() {
```

Comment:
If we adding log message for pollRule method, should we also add log message for pollManifest method?

Reply:
Added.